fix download filepath bug

webern opened the pull request:
*Issue #, if available:*

Closes #38 

*Description of changes:*

This fixes a bug where downloading to /some/path/file.bin was not respected. only the filename was considered and file.bin would be placed in the local directory.

I also changed the name of the positional argument `filename` in both download and upload to `file` which feels like a canonical and accurate way to say that we are accepting a path.

In `src/download.rs` at (currently, line 60), I'm not sure why we validate that the file_name is UTF-8, I could remove that check if we want.

*Testing*

```sh
cargo run -- download snap-xxxf0xxx349xxxab1 /tmp/snap.bin
cargo run -- upload  /tmp/snap.bin
```

The above commands worked (I was not running from /tmp) and previously they would not.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

webern committed:
fix download filepath bug

This fixes a bug where downloading to /some/path/file.bin was not
respected. only the filename was considered and file.bin would be placed
in the local directory.

tjkirch commented on src/bin/coldsnap/main.rs:
```diff
@@ -159,7 +159,7 @@ struct DownloadArgs {
 
     #[argh(positional)]
-    filename: PathBuf,
+    file: PathBuf,
```
> `file` which feels like a canonical and accurate way to say that we are accepting a path

Why not `path`?  The file is the object, the path is its address.

```suggestion
    path: PathBuf,
```

webern replied:
I chose file because I thought it was better for the usage text:

```
Usage: coldsnap download <snapshot_id> <file> [--force] [--no-progress]

Download an EBS snapshot into a local file.

Options:
  --force           overwrite an existing file
  --no-progress     disable the progress bar
  --help            display usage information
```

Path would be fine, I guess, but it felt like `file` was more canonical for command line program usage statements.

tjkirch reviewed (approved):
Mind splitting the bug fix and the renaming commits?

zmrow reviewed (approved):
Thanks for doing this!

:muscle:

webern committed:
rename positional argument from filename to file

The positional argument for the download and upload commands was named
"filename" which was misleading since it it actually a filepath. This is
now renamed to "file" which is a more common way for a command line
program to indicate that it accepts the path to a file.

webern commented:
> webern force-pushed the webern:filepath branch from b1a30b1 to 2eca44d

Split into two commits.

webern commented:
> webern force-pushed the webern:filepath branch from 2eca44d to d4460d9

rebase after merging tokio v1 stuff
